addons/account/models/account_account.py

```diff
@@ -351,7 +354,7 @@ def _check_used_as_journal_default_debit_credit_account(self):
     @api.constrains('code')
     def _check_account_code(self):
         for account in self:
-            if account.code and not re.match(ACCOUNT_CODE_REGEX, account.code):
+            if account.code and not re.match(ACCOUNT_CODE_REGEX, account.code) and self.env.company in account.company_ids:
```

Comment:
Not sure to get why we need this :thinking:

Reply:
This is to enable users to create more expressive 'account codes' for consolidation. The idea is that you create a company just for consolidation, and assign a code in that company to each of your existing accounts. By enabling users to put whatever they want as the code, this effectively gives them flexibility to specify any header they want when generating a consolidated report where they are grouping by 'account code' from the perspective of the consolidation company.

(yes, this is a poor version of the mapping - shhhh....) :smile: 

In any case this was requested by @tsb-odoo :smile: